sw/device/tests/i2c_host_irq_test.c

```diff
@@ -207,6 +207,26 @@ static status_t rx_threshold_irq(void) {
   return OK_STATUS();
 }
 
+static status_t rx_overflow_irq(void) {
+  TRY(dif_i2c_irq_set_enabled(&i2c, kDifI2cIrqRxOverflow, kDifToggleEnabled));
+
+  irq_global_ctrl(false);
+  irq_fired = UINT32_MAX;
+  irq_global_ctrl(true);
+
+  const uint8_t kAddr[2] = {0x03, 0x21};
+  uint8_t bytes[I2C_PARAM_FIFO_DEPTH - 1];
```

Comment:
I am a bit confused: the array is less than the fifo depth? Is there anything else that goes into the RX FIFO that triggers the overflow?

Reply:
Exccelent observation! The irq was pending due a previous test (nak) which was erroneous reading 0 bytes (for the HW 0 = FIFO_DEPTH).
So, I fixed the nak test to read 1 byte. Fixed the rx_overflow to read `I2C_PARAM_FIFO_DEPTH+1`.